Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1457,8 +1461,12 @@ void HTMLElement::popoverAttributeChanged(const AtomString& value)
         { CSSSelector::PseudoClassClosed, newPopoverState != PopoverState::None }
     });
 
-    if (oldPopoverState != PopoverState::None)
-        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::No);
+    if (oldPopoverState != PopoverState::None) {
+        auto oldValue = attributeWithoutSynchronization(popoverAttr);
+        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::Yes, SuppressExceptions::Yes);
+        if (!isConnected() || !popoverData() || oldValue != attributeWithoutSynchronization(popoverAttr))
+            return;
+    }
```

Comment:
This doesn't match the spec either. I have a slight preference for what the spec does of not firing the events (which as the results means the type cannot be changed at this step). Can we update the WPT? cc @annevk 

https://html.spec.whatwg.org/#the-popover-attribute

<img width="914" alt="image" src="https://user-images.githubusercontent.com/1795135/224180595-8bc64304-729f-4f83-8a8c-a76505e7dcf8.png">

Reply:
If you look at the pasted snippet, the spec seems to say focusPreviousElement=true, fireEvents=true and throwExceptions=false, so it does seem to match the current spec?

Reply:
Yeah, I think you're correct. I don't really like attribute changes firing events synchronously, but isn't the end of the world either I suppose. Chromium follows the specification I assume?

Reply:
Yes, chromium follows the spec here, i.e. fires the event.